### ISSUE TITLE
[icu] Build error

### ISSUE THREAD

**Issue by user426** [microsoft/vcpkg#24140]
**Debian 11**
- OS: /Linux

**To Reproduce**
`./vcpkg install icu[core]:x64-linux`

**Failure logs**
```
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:128 (message):
    Command failed: /usr/bin/bash -c "V=1 ./../src/c-70_1-src-a6ee276eef.clean/source/configure  \"--disable-samples\" \"--disable-tests\" \"--disable-layoutex\" \"--disable-silent-rules\" \"--verbose\" \"--disable-shared\" \"--enable-static\" \"--enable-debug\" \"--disable-release\" \"--prefix=/home/me/.local/bin/vcpkg/installed/x64-linux/debug\" \"--bindir=\\${prefix}/../tools/icu/debug/bin\" \"--sbindir=\\${prefix}/../tools/icu/debug/sbin\" \"--libdir=\\${prefix}/lib\" \"--includedir=\\${prefix}/../include\" \"--datarootdir=\\${prefix}/share/icu\""
```
**Log from file vcpkg/buildtrees/icu/config-x64-linux-dbg-out.log :**
```
checking for ICU version numbers... release 70.1, library 70.1, unicode version 14.0
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking whether to build debug libraries... yes
checking whether to build release libraries... no
checking for clang... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether clang accepts -g... yes
checking for clang option to accept ISO C89... none needed
checking for clang++... clang++
checking whether we are using the GNU C++ compiler... yes
checking whether clang++ accepts -g... yes
checking pkg-config is at least version 0.20... yes
checking for icu-le-hb... no
checking how to run the C preprocessor... clang -E
checking for a BSD-compatible install... /usr/bin/install -c
checking for python3... /usr/bin/python3
checking for gmake... /usr/bin/gmake
checking for doxygen... /usr/bin/doxygen
checking checking for executable suffix... 
checking whether strict compiling is on... yes
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking size of void *... 8
checking whether runnable 64 bit binaries are built by default... yes
checking which Makefile fragment to use for x86_64-pc-linux-gnu... mh-linux
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking whether to enable auto cleanup of libraries... no
checking whether to enable draft APIs... yes
checking for ranlib... ranlib
checking for a sed that does not truncate output... /usr/bin/sed
checking for ar... ar
checking whether to enable renaming of symbols... yes
checking whether to enable function and data tracing... no
checking elf.h usability... yes
checking elf.h presence... yes
checking for elf.h... yes
checking whether to enable dynamic loading of plugins. Ignored if plugins disabled.... yes
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking for library containing dlopen... -ldl
checking for dlopen... yes
checking for gettimeofday... yes
checking if we have a C++ compiler... Good
checking if we have a C++11 compiler... yes
configure: Adding CXXFLAGS option -std=c++11
checking for pthread_attr_init in -lpthread... yes
checking for mmap... yes
checking for genccode assembly... -a gcc
checking for inttypes.h... (cached) yes
checking dirent.h usability... yes
checking dirent.h presence... yes
checking for dirent.h... yes
checking whether byte ordering is bigendian... no
checking for nl_langinfo... yes
checking for nl_langinfo's argument to obtain the codeset... CODESET
checking for namespace support... yes
checking for properly overriding new and delete... yes
checking for placement new and delete... yes
checking for popen... yes
checking for tzset... yes
checking for tzname... yes
checking for timezone... yes
checking for strtod_l... yes
checking xlocale.h usability... no
checking xlocale.h presence... no
checking for xlocale.h... no
checking for std::string_view... no
checking for int8_t... yes
checking for uint8_t... yes
checking for int16_t... yes
checking for uint16_t... yes
checking for int32_t... yes
checking for uint32_t... yes
checking for int64_t... yes
checking for uint64_t... yes
checking wchar.h usability... yes
checking wchar.h presence... yes
checking for wchar.h... yes
checking for library containing wcscpy... none required
checking size of wchar_t... 4
checking for UTF-16 string literal support... available
checking for a library suffix to use... none
```

**Log from file vcpkg/buildtrees/icu/config-x64-linux-dbg-err.log :**
```
configure: WARNING: unrecognized options: --disable-silent-rules
./../src/c-70_1-src-a6ee276eef.clean/source/configure: line 8012: syntax error near unexpected token `newline'
./../src/c-70_1-src-a6ee276eef.clean/source/configure: line 8012: `AX_CHECK_COMPILE_FLAG('
```

**Comment by bold84** [comment]
I have the same issue on Mac M1

**Comment by autoantwort** [comment]
Maybe this helps: https://stackoverflow.com/questions/30412576/autotools-syntax-error-with-ax-check-compile-flag

**Comment by user426** [comment]
> I have the same issue on Mac M1

try installing autoconf-archive on MAC OS

**Comment by daschuer** [comment]
Can we reopen this issue, to issue a reasonable error message? 
It took me quite a while to find this closed bug. A huge waste of time.

**Comment by JackBoosY** [comment]
Fixed by #24512.
